src/state/slices/marketDataSlice/selectors.ts

```diff
@@ -17,34 +17,36 @@ export const selectCryptoMarketData = (state: ReduxState) => state.marketData.cr
 export const selectCryptoMarketDataIds = (state: ReduxState) => state.marketData.crypto.ids
 const selectFiatMarketData = (state: ReduxState) => state.marketData.fiat.byId
 
-export const selectMarketDataSortedByMarketCap = createDeepEqualOutputSelector(
+export const selectSelectedCurrencyMarketDataSortedByMarketCap = createDeepEqualOutputSelector(
   selectCryptoMarketData,
+  selectCryptoMarketDataIds,
   selectFiatMarketData,
   selectSelectedCurrency,
-  (cryptoMarketData, fiatMarketData, selectedCurrency): MarketDataById<AssetId> => {
+  (
+    cryptoMarketData,
+    marketDataAssetIds,
+    fiatMarketData,
+    selectedCurrency,
+  ): MarketDataById<AssetId> => {
     const fiatPrice = bnOrZero(fiatMarketData[selectedCurrency]?.price ?? 1) // fallback to USD
-    return (
-      Object.entries(cryptoMarketData)
-        // apply fiat conversion
-        .map<[AssetId, MarketData]>(([assetId, assetMarketData]) => {
-          return [
-            assetId,
-            {
-              ...assetMarketData,
-              price: bnOrZero(assetMarketData?.price).times(fiatPrice).toString(),
-              marketCap: bnOrZero(assetMarketData?.marketCap).times(fiatPrice).toString(),
-              volume: bnOrZero(assetMarketData?.volume).times(fiatPrice).toString(),
-              changePercent24Hr: assetMarketData?.changePercent24Hr ?? 0,
-            },
-          ]
-        })
-        // sort by market cap
-        .sort(([, a], [, b]) => (bnOrZero(a.marketCap).lt(b.marketCap) ? 1 : -1))
-        .reduce<MarketDataById<AssetId>>((acc, [assetId, assetMarketData]) => {
-          acc[assetId] = assetMarketData
+    // No conversion needed
+    return selectedCurrency === 'USD'
+      ? cryptoMarketData
+      : marketDataAssetIds.reduce<MarketDataById<AssetId>>((acc, assetId) => {
+          const assetMarketData = cryptoMarketData[assetId]
+          // Market data massaged to the selected currency
+          const selectedCurrencyAssetMarketData = {
+            ...assetMarketData,
```

Comment:
could be mutaty for even better perf but not a big deal IMO - perhaps `Object.assign()` does make more sense however

Reply:
dcc8efbc3